### PR TITLE
Update check-update to pull from metadata.json

### PR DESCRIPTION
**Pull request by misscoded**
###  Summary

This PR updates the `check-update` hook to pull version information directly from the `metadata.json` that's published at https://api.slack.com/slackcli/metadata.json.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/{project_slug}/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: src/check_update.ts [hunk 1]
```diff
@@ -284,7 +282,7 @@ export function createUpdateResp(
       // Add the dependency that failed to be fetched to the top-level error message
       if (sdk.error && sdk.error.message) {
         errorMsg += errorMsg
-          ? `, ${sdk}`
+          ? `, ${sdk.name}`
```
Comment (misscoded) [inline src/check_update.ts:285]: Ran into a bug along the way! 🐛

File: src/tests/check_update_test.ts [hunk 2]
```diff
@@ -211,27 +211,42 @@ Deno.test("check-update hook tests", async (t) => {
...
-          return new Response(null, {
-            headers: { location: "/x/slack@0.1.1" },
-          });
+        mockFetch.mock("GET@/slackcli/metadata.json", (_req: Request) => {
+          return new Response(mockMetadataJSON);
         });
-        const version = await fetchLatestModuleVersion("slack");
-        assertEquals(version, "0.1.1", "inocrrect version returned");
+        const version = await fetchLatestModuleVersion("deno-slack-sdk");
+        assertEquals(version, "1.3.0", "incocrrect version returned");
```
Comment (misscoded) [inline src/tests/check_update_test.ts:249]:
```suggestion
        assertEquals(version, "1.3.0", "incorrect version returned");
```

🤦🏼‍♀️

**Review by cchensh — APPROVED** [review]
Thank you for making this change @misscoded! It looks good to me, I tested locally by changing metadata.json to my public Git (bump `deno_slack_api` version to be higher than prod) and everything looks good!:

image

**Review by filmaj — APPROVED** [review]
LGTM! Worked like a charm:

```
🛠️  An update from the Slack SDK is available:

    › deno_slack_sdk
      1.2.0 → 1.3.0

    › deno_slack_api
      1.1.0 → 1.2.0

   For more information about this update, visit:
   https://api.slack.com/future/changelog

? 🚀 Do you want to auto-update to the latest versions now? Yes
   Starting the auto-update...
   Wrapping up updates...

✨ The following dependencies have been updated:

    ✔ deno_slack_sdk
      1.2.0 → 1.3.0

    ✔ deno_slack_api
      1.1.0 → 1.2.0

   Please review any file changes before continuing development.
```

**Pull request merged**